aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,116 @@
+
+>[!NOTE]
+> If you place Static File Middleware before Response Caching Middleware, you can provide a prefix to the middleware that will force it to ignore requests with a path that doesn't match your prefix. For example, you can prefix your application's static file links with **assets** (`http://www.myapp.com/assets/staticfile.txt`) and configure the middleware to ignore any requests that do not include the **assets** prefix, `app.UseStaticFiles("/assets")`. Note that "**assets**" is only part of the path in your links to the static files. Your application will not have an **assets** folder, and your static files should remain in the **wwwroot** folder of your deployed application (**wwwroot\staticfile.txt**).
+
+> The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), will be ignored by the middleware.
+
+## Options
+The middleware offers two options for controlling response caching.
+
+Option | Default Value
+--- | ---
+UseCaseSensitivePaths | <p>Determines if responses will be cached on case-sensitive paths.</p><p>The default value is `false`.</p>
+MaximumBodySize | <p>The largest cacheable size for the response body in bytes.</p>The default value is `64 * 1024 * 1024` [64 MB (67,108,864 bytes)].</p>
+
+The following example configures the middleware to cache responses smaller than or equal to 1 MB (1,024 bytes) using case-sensitive paths, storing the responses to `/page1` and `/Page1` separately.
+
+```csharp
+services.AddResponseCaching(options =>
+{
+    options.UseCaseSensitivePaths = true;
+    options.MaximumBodySize = 1024;
+});
+```
+
+## `ResponseCache` attribute
+The `ResponseCache` attribute specifies the parameters necessary for setting appropriate headers for response caching. The only parameter of the `ResponseCache` attribute that strictly requires the middleware is `VaryByQueryKeys`, which does not correspond to an actual HTTP header. See [ResponseCache Attribute](response.md#responsecache-attribute) and the [ResponseCacheAttribute Class API Reference](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute) for usage and a description of the parameters.
+
+## `VaryByQueryKeys` without using the `ResponseCache` attribute
+You can vary response caching with the `VaryByQueryKeys` feature without the `ResponseCache` attribute by using the `ResponseCachingFeature` directly from the `IFeatureCollection` of the `HttpContext`.
+
+```csharp
+var responseCachingFeature = context.HttpContext.Features.Get<IResponseCachingFeature>();
+if (responseCachingFeature != null)
+{
+    responseCachingFeature.VaryByQueryKeys = new[] { "MyKey" };
+}
+```
+
+## HTTP response caching headers
```

Comment:
"HTTP headers used by Response Caching Middleware" since some of these headers like `If-None-Match` is a request header and `Cache-Control` is bidirectional.